### PR TITLE
Added foot note plus several enhancements and code improvements

### PR DESCRIPTION
**Pull request by ema2159**
Created a customizable footnote with its own customizable icon. Respected the most I could @seagle0128 implementation in Centaur to try and not mess up his cool config. It looks like this:
DeepinScreenshot_select-area_20190531015917
Also, improved a little bit the order of the code with things like for example generalizing the banner title, init info and foot note centering with a function.

Hope you like it and approve it! Please test it

### CODE REVIEW

**Comment by seagle0128** [comment]
It looks fine except the minor issue I comment above.

Another suggestion is to add one more customization `dashboard-navigator` or something like it, and put it above `dashboard-init-info`, then the users can add some buttons or links there.

**Comment by seagle0128** [comment]
It looks fine with me except one minor issue above. Thank you!

**Comment by ema2159** [comment]
What issue? Emmacs? That's not an error jaja, my name is Emmanuel, Emma in short so I like to mess calling Emacs Emmacs. 

By the way I also modified the text banners face to match the headings and also enabled the init info for it which wasn't enabled.

I tried to use your center-line function but I couldn't understand how it worked so I used the piece of code that was originally on dashboard to center the content as the center-line function calling it before instead of after the centered text.

**Comment by seagle0128** [comment]
Sorry! I started a code-review but forgot to submit the comments.

`(defvar dashboard-footer-icon (all-the-icons-fileicon ...)`
> Should consider TTY. This may cause an exception.

Regarding `dashboard-center-line`, it's a little bit different from yours. It gets the actual width of this line, and inserts some spaces at the beginning. The reimplementation is to calculate with the string width. It's fine unless you don't know the string. In centaur navigator, I create some widget buttons, and I cannot get the string so use `current-column` to calculate the accurate width.

File: dashboard-widgets.el [hunk 1]
```diff
@@ -88,6 +92,15 @@ to the specified width, with aspect ratio preserved."
                                     (length package-activated-list) (emacs-init-time))
   "Init info with packages loaded and init time.")
 
+(defvar dashboard-footer "The one true editor, Emacs!"
+  "A footer with some short message.")
+
+(defvar dashboard-footer-icon (all-the-icons-fileicon "emacs"
```
[the remark this answers is not in the document]
Comment (ema2159) [reply dashboard-widgets.el]: I was lost with what issue you were talking about jaja! A simple `(when display-graphic-p)` in that defvar should do it right? I was about to keep it like in your config but I couldn't find a decent replacement for the emacs icon

**Comment by ema2159** [comment]
By the way regarding the dashboard-navigator customization I was thinking about implementing it and came with the idea that the user passes an alist with the icon they want for the button, the text they want in it and the internal/external link that button accesses and that dashboard automatically built the navigator. I don't now how to make a "triple" alist though. Would you help me with an idea?

**Comment by seagle0128** [comment]
@ema2159 `dashboard-navigator` should contain `tag` (including text and/or icon?), `action`, `help-echo`, etc. I think the list below could address this issue.

```elisp
(defvar dashboard-navigator-alist
  '(("tag1" action1 "help1" face1 ...)
    ("tag2" action2 "help2" face2 ...)))
```
`car`, `cdr` etc. can access the items.

The essential items should be tag and action, so it could be

```elisp
(defvar dashboard-navigator-alist
  '(("tag1" . action1)
    ("tag2" . action2)))
```

Does that address your issue?

**Comment by ema2159** [comment]
> @ema2159 `dashboard-navigator` should contain `tag` (including text and/or icon?), `action`, `help-echo`, etc. How about this list?
> 
> ```emacs-lisp
> (defvar dashboard-navigator-alist
>   '(("tag1" action1 "help1" face1 ...)
>     ("tag2" action2 "help2" face2 ...)))
> ```
> 
> `car`, `cdr` etc. can access the items.
> 
> The essential items should be tag and action, so it could be
> 
> ```emacs-lisp
> (defvar dashboard-navigator-alist
>   '(("tag1" . action1)
>     ("tag2" . action2)))
> ```
> 
> The widget style of the buttons is `push-button` or `item`.
> 
> Does that address your issue?

Yes! I'll try that, that'll make Dashboard a really complete and aesthetic package, more than it is right now. From there I don't know what enhancements can be done.

**Comment by ema2159** [comment]
@JesusMtnez I think this PR is good to go!

**Comment by JesusMtnez** [comment]
Thank you @ema2159 !! This is a really good job. I'll be busy tomorrow, but it'll try to merge it on Sunday or on Monday.

**Comment by ema2159** [comment]
@JesusMtnez no hay prisa! Do it when you can!

**Comment by JesusMtnez** [comment]
Do you mind moving this last commit to it's own PR?

**Comment by ema2159** [comment]
Of course! It's only two lines tho so is it worth it?

**Comment by JesusMtnez** [comment]
Thank you @ema2159 !!

**Pull request merged**